blog-posts/middleware-on-redux/middleware-on-redux.md

```diff
@@ -0,0 +1,81 @@
+---
+published: false
+title: 'Middleware on redux'
+cover_image: 'https://raw.githubusercontent.com/ramclen/documenting-dev/master/blog-posts/middleware-on-redux/assets/redux.png'
+description: 'Answering questions about middleware on redux'
+tags: redux, middleware, typescript, javascript
+series:
+canonical_url:
+---
+
+If you are start learning Redux, middlewares could be a pain know what it is and why we need them... So I am going to try to give you few keys (and links) to help you in your learning process.
+
+## What is a middleware?
+
+A middleware is usually called "Software glue". It is usually a piece of software that it is in the middle of two software, and it make easy the communication between them.
+
+A middleware become popular with the beginning of Operating systems architecture. It made easy the way that engineers get/set data to operating system.
+
+In [redux documentation](https://redux.js.org/advanced/middleware/) says:
+
+`Provides a third-party extension point between dispatching an action, and the moment it reaches the reducer.`
+
+So is basically a piece of code between actions and reducers.
+
+## Why we need them?
+
+Yes, I was thinking the same, why we have to complicate things that much? But hold on, it have good a reasons.
+
+Actually, It is technically possible do it without middlewares. But, as a rule of thumb is a bad idea and you could mess things.
+
+Redux try control access to state and reduce any problems. For that propose they try to give you few guidelines. They usually are aimed to reduce complexity and organize everything by single responsibility. The things that clarify me this were:
+
+1. It made easy the process to create new kind of actions. The big thing here is we are going to probably send the dispatch to our new action. That have two disadvantages:
+
+   1. Made the method signature complex and a bit difficult to read.
+   2. Now we have two different ways to call actions.
+
+2. We encapsulate this complexity from our UI Components. In that way every action is executed in the same way. If in the future we change that actions and it does not need any middleware Components does not need to know or change anything.
```

Comment:
"If we change that actions in the future" sounds better.
Add a comma after "middleware"